contracts/liquidator/LiquidatorV2.sol

```diff
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "./vendor/@uniswap/v3-core/contracts/interfaces/callback/IUniswapV3FlashCallback.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/base/PeripheryPayments.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/base/PeripheryImmutableState.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/PoolAddress.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/CallbackValidation.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+
+import "../CometInterface.sol";
+import "../ERC20.sol";
+
+/**
+ * @title Liquidator contract compatible with 1inch DEX aggregator
+ * @author Compound
+ */
+contract LiquidatorV2 is IUniswapV3FlashCallback, PeripheryImmutableState, PeripheryPayments {
+    /** Errors */
+    error InsufficientBalance(uint256 available, uint256 required);
+    error Unauthorized();
+
+    /** Events **/
+    event Absorb(address indexed initiator, address[] accounts);
+    event AbsorbWithoutBuyingCollateral();
+    event Pay(address indexed token, address indexed payer, address indexed recipient, uint256 value);
+    event BuyAndSwap(
+        address indexed tokenIn,
+        address indexed tokenOut,
+        uint256 baseAmountPaid,
+        uint256 assetBalance,
+        uint256 amountOut
+    );
+
+    /// @notice Compound Comet protocol
+    CometInterface public immutable comet;
```

Comment:
alternatively, we could take in a comet instance as an argument, similar to the way Bulker does

Reply:
I do like the idea of having `Comet` be part of the calldata. Shouldn't be much gassier but offers more flexibility.

Reply:
updated to take Comet as an argument